From 2db577a5b4f765cfed1d6bdd89c8bc5501801018 Mon Sep 17 00:00:00 2001
From: Alessandro Chitolina <alekitto@gmail.com>
Date: Sun, 6 Oct 2024 01:47:23 +0200
Subject: [PATCH] fix: support postgresql 17

---
 Makefile                     | 3 ++-
 src/monitor/version_compat.h | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 93abdfd..e73827b 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 .DEFAULT_GOAL := all
 
 # Supported PostgreSQL versions:
-PGVERSIONS = 11 12 13 14 15 16
+PGVERSIONS = 11 12 13 14 15 16 17
 
 # Default version:
 PGVERSION ?= $(lastword $(PGVERSIONS))
@@ -278,6 +278,7 @@ BUILD_ARGS_pg13 = --build-arg PGVERSION=13 --build-arg CITUSTAG=v10.2.9
 BUILD_ARGS_pg14 = --build-arg PGVERSION=14 --build-arg CITUSTAG=$(CITUSTAG)
 BUILD_ARGS_pg15 = --build-arg PGVERSION=15 --build-arg CITUSTAG=$(CITUSTAG)
 BUILD_ARGS_pg16 = --build-arg PGVERSION=16 --build-arg CITUSTAG=$(CITUSTAG)
+BUILD_ARGS_pg17 = --build-arg PGVERSION=17 --build-arg CITUSTAG=$(CITUSTAG)
 
 # DOCKER BUILDS
 
diff --git a/src/monitor/version_compat.h b/src/monitor/version_compat.h
index 75555c9..9b31254 100644
--- a/src/monitor/version_compat.h
+++ b/src/monitor/version_compat.h
@@ -14,8 +14,8 @@
 
 #include "postgres.h"
 
-/* we support Postgres versions 10, 11, 12, 13, 14, 15, and 16. */
-#if (PG_VERSION_NUM < 100000 || PG_VERSION_NUM >= 170000)
+/* we support Postgres versions 10, 11, 12, 13, 14, 15, 16 and 17. */
+#if (PG_VERSION_NUM < 100000 || PG_VERSION_NUM >= 180000)
 #error "Unknown or unsupported postgresql version"
 #endif
 
-- 
2.34.1

